restic-rest-server: 0.12.1 -> 0.13.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://github.com/restic/rest-server/releases.


meta.description for restic-rest-server is: High performance HTTP server that implements restic's REST backend API

meta.homepage for restic-rest-server is: https://github.com/restic/rest-server

meta.changelog for restic-rest-server is: https://github.com/restic/rest-server/blob/v0.13.0/CHANGELOG.md


###### Updates performed
- Golang update

###### To inspect upstream changes

- [Release on GitHub](https://github.com/restic/rest-server/releases/tag/v0.13.0)

- [Compare changes on GitHub](https://github.com/restic/rest-server/compare/v0.12.1...v0.13.0)

###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0.13.0 with grep in /nix/store/hwvidk7wsw5p2mb0ixagm3yn9s2si76l-restic-rest-server-0.13.0
- found 0.13.0 in filename of file in /nix/store/hwvidk7wsw5p2mb0ixagm3yn9s2si76l-restic-rest-server-0.13.0

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2 total rebuild path(s)

2 package rebuild(s)

First fifty rebuilds by attrpath

restic-rest-server
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/hwvidk7wsw5p2mb0ixagm3yn9s2si76l-restic-rest-server-0.13.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A restic-rest-server https://github.com/r-ryantm/nixpkgs/archive/08c791cc3545bb474c660f404a92e8637ab360e5.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/08c791cc3545bb474c660f404a92e8637ab360e5#restic-rest-server
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/hwvidk7wsw5p2mb0ixagm3yn9s2si76l-restic-rest-server-0.13.0
ls -la /nix/store/hwvidk7wsw5p2mb0ixagm3yn9s2si76l-restic-rest-server-0.13.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>restic-rest-server</li>
  </ul>
</details>

---

###### Maintainer pings

cc @dotlambda for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

NickCao reviewed (commented):
The nixos module might need an update.

dotlambda committed:
restic-rest-server: 0.12.1 -> 0.13.0

Diff: https://github.com/restic/rest-server/compare/v0.12.1...v0.13.0

Changelog: https://github.com/restic/rest-server/blob/v0.13.0/CHANGELOG.md

NickCao reviewed (approved):
I was thinking about socket activation, but it turns out we were already doing this. Thanks!
